meson.build

```diff
@@ -0,0 +1,34 @@
+project('lgi', 'c',
+  version: '0.9.2',
+  meson_version: '>= 0.44.0',
+  default_options: [
+    'warning_level=2',
+    'buildtype=debugoptimized',
+  ],
+)
+
+lua_name = get_option('lua-pc')
+lua_dep = dependency(lua_name)
```

Comment:
Would it make sense to replace this option with a brute force search for all the possible options that were seen "in the wild"?

Reply:
Well, I won't maintain that list. If you will then its up to you.

Reply:
I think the list would just be `lua, lua5.1, lua51, lua5.2, lua52, lua5.3, lua53`. That does not seem too bad to me, but perhaps it is nicer to require people to specify this explicitly. That way it is easier to "force switch" to another Lua version.
Is it possible to customise the error message here to tell people how to specify this? E.g., how about changing the default value of `lua-pc` to "unspecified", if this is the value seen in `meson.build`, then something like `warning("You can specify the name of the Lua pkg-config file via -Dlua-pc=lua53")` is done (and perhaps then a default of `lua`) is used?
I had to Google to figure out how to set this option. Someone who tries to just run `meson` won't even know that this option exist.

Does either of that make sense?

Also: Please ping me if you think everything is addressed and I will likely just merge this without taking a second look.

CC @ILMostro just FYI and perhaps this helps simplify building on Gentoo (at the expense of depending on meson).

Reply:
Sure we can do that.